common/stats/verify_stats.go

```diff
@@ -0,0 +1,108 @@
+package stats
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"runtime"
+	"strings"
+	"testing"
+)
+
+/*
+Utilities for validating the stats registry contents
+*/
+/*
+add new Checker functions here as needed
+*/
+/*
+errors if a is not float64, returns true if a == b
+*/
+func FloatEqTest(a, b interface{}) bool {
+	if b == nil && a == nil {
+		return true
+	}
+	aflt := a.(float64)
+	bflt := b.(float64)
+	return aflt == bflt
+}
+
+/*
+errors if a is not float64, returns true if a > b
+*/
+func FloatGTTest(a, b interface{}) bool {
+	if b == nil && a == nil {
+		return true
+	}
+	aflt := a.(float64)
+	bflt := b.(float64)
+	return aflt > bflt
+}
+
+/*
+errors if a is not int64, returns true if a == b
+*/
+func Int64EqTest(a, b interface{}) bool {
+	if b == nil && a == nil {
+		return true
+	}
+	aint := a.(int64)
+	bint := b.(int)
+	return aint == int64(bint)
+}
+func DoesNotExist(a, b interface{}) bool {
+	return a == nil
+}
+
+/*
+defines the condition checker to use to validate the measurement.  Each Checker(a, b) implementation
+will expect a to be the 'got' value and b to be the 'expected' value.
+*/
+type Rule struct {
+	Checker func(interface{}, interface{}) bool
+	Value   interface{}
+}
+
+/*
+Verify that the stats registry object contains values for the keys in the contains map parameter and that
+each entry conforms to the rule (condition) associated with that key.
+*/
+func VerifyStats(tag string, statsRegistry StatsRegistry, t *testing.T, contains map[string]Rule) {
+
+	asFinagleRegistry, ok := statsRegistry.(*finagleStatsRegistry)
+	err := false
+	var msg bytes.Buffer
+	msg.WriteString(tag)
+	msg.WriteString(":stats registry error:\n")
+
+	if ok {
+		asJson := asFinagleRegistry.MarshalAll()
+
+		for key, rule := range contains {
+			gotValue, ok := asJson[key]
+			if !ok {
+				if !strings.Contains(runtime.FuncForPC(reflect.ValueOf(rule.Checker).Pointer()).Name(), "DoesNotExist") {
```

Comment:
It's generally better if we can avoid reflection - can you try assigning DoesNotExist and the other checkers to their own variables and then operate on pointers to those variables? That way you can do plain equality.